esmvalcore/_task.py

```diff
@@ -279,12 +280,51 @@ def __repr__(self):
         return f"{self.__class__.__name__}({repr(self.name)})"
 
 
+class ResumeTask(BaseTask):
+    """Task for re-using preprocessor output files from a previous run."""
+
+    def __init__(self, prev_preproc_dir, preproc_dir, name):
+        """Create a resume task."""
+        # Set the path to the file resulting from running this task
+        self._metadata_file = preproc_dir / 'metadata.yml'
+
+        # Reconstruct output
+        prev_metadata_file = prev_preproc_dir / 'metadata.yml'
+        with prev_metadata_file.open('rb') as file:
+            prev_metadata = yaml.safe_load(file)
+
+        products = set()
+        for prov_filename, attributes in prev_metadata.items():
+            # Update the filename in case the output directory was moved
+            # since the original run
+            filename = str(prev_preproc_dir / Path(prov_filename).name)
+            attributes['filename'] = filename
+            product = TrackedFile(filename,
+                                  attributes,
+                                  prov_filename=prov_filename)
+            products.add(product)
+
+        super().__init__(ancestors=None, name=name, products=products)
+
+    def _run(self, _):
+        """Return the result of a previous run."""
+        metadata = self.get_product_attributes()
+
+        # Write metadata to file
+        self._metadata_file.parent.mkdir(parents=True)
+        with self._metadata_file.open('w') as file:
+            yaml.safe_dump(metadata, file)
+
+        return [str(self._metadata_file)]
```

Comment:
nice minimal class! Can this be run once for all the upcoming preprocessing (parallel) tasks and not multiple times of it?

Reply:
All tasks that are created from a recipe are run exactly once

Reply:
what I meant was run a single overarching task for all the preprocessing tasks, I know those are individual tasks and run once. Anyways, not for this PR, for future enhancements etc